src/main/java/training/subsystems/RobotArm/roller/Roller.java

```diff
@@ -0,0 +1,47 @@
+package training.subsystems.RobotArm.roller;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import utils.GBSubsystem;
+
+public class Roller extends GBSubsystem {
+
+	private CANSparkMax motor;
+	private static Roller instance;
+
+	private Roller() {
+		this.motor = new CANSparkMax(RollerConstants.MOTOR_ID, CANSparkLowLevel.MotorType.kBrushless);
+		motor.getPIDController().setP(RollerConstants.KP);
+		motor.getPIDController().setI(RollerConstants.KI);
+		motor.getPIDController().setD(RollerConstants.KD);
+	}
+
+	public static Roller getInstance() {
+		if (instance != null)
+			instance = new Roller();
+		return instance;
+	}
+
+	public void moveAtSpeed(double velocity) {
+		motor.getPIDController().setReference(velocity, CANSparkBase.ControlType.kVelocity, RollerConstants.PID_SLOT);
+	}
+
+
+	public Rotation2d getPower() {
+		return Rotation2d.fromDegrees(motor.getEncoder().getPosition());
+	}
+
+	public boolean isAtSpeed(Rotation2d velocity) {
+		return (getPower() == velocity);
```

Comment:
within a tolerance... Don't use == to compare real-life values; there would always be a error.